### PR TITLE
Add WDL builtin test 'sub'.

### PR DESCRIPTION
**Pull request by w-gao**
#3167.

### CODE REVIEW

File: src/toil/test/wdl/builtinTest.py [hunk 1]
```diff
@@ -234,6 +243,14 @@ def check_function(self, function_name, cases, expected_result):
                 self.assertEqual(result, expected_result)
                 shutil.rmtree(output_dir)
 
+    def test_sub(self):
+        # this workflow swaps the extension of a TSV file to CSV, with String and File inputs.
+        self.check_function('sub', cases=['as_input'], expected_result='src/toil/test/wdl/test.csv')
+
+        # NOTE: the result differs from Cromwell since we copy the file to the file store without
+        # preserving the path. Cromwell would return 'src/toil/test/wdl/test.csv' instead.
+        self.check_function('sub', cases=['as_input_with_file'], expected_result='test.csv')
```
Comment (w-gao) [inline src/toil/test/wdl/builtinTest.py:252]: I don't think we can produce the same behavior with our current implementation?

File: src/toil/wdl/wdl_functions.py [hunk 2]
```diff
@@ -228,15 +229,24 @@ def process_infile(f, fileStore):
         raise RuntimeError('Error processing file: '.format(str(f)))
 
 
-def sub(a, b, c):
-    if isinstance(a, tuple):
-        a = a[1]
-    if isinstance(a, tuple):
```
Comment (w-gao) [inline src/toil/wdl/wdl_functions.py:234]: I assume `a` is a typo?

**Review by w-gao — COMMENTED** [review]
Apparently I'm reviewing my own PR.. I added some questions that came up during development.

File: src/toil/wdl/wdl_functions.py (hunk 2 above)
Comment (DailyDreaming) [reply src/toil/wdl/wdl_functions.py:234]: Yes, it is.

File: src/toil/test/wdl/builtinTest.py (hunk 1 above)
Comment (DailyDreaming) [reply src/toil/test/wdl/builtinTest.py:252]: Oh, interesting, you're right.  I don't think we need to address this immediately, and the comment should be enough for now, especially since `sub()` is still doing exactly what it should.

**Review by DailyDreaming — APPROVED** [review]
LGTM.  I don't think that we can account for Toil's path mangling right now.  Thanks for pointing it out and adding a test appropriate to the edge case!
